AFQ/api/bundle_dict.py

```diff
@@ -1138,3 +1099,67 @@ def __add__(self, other):
             self.resample_to,
             self.resample_subject_to,
             self.keep_in_memory)
+
+
+def apply_to_roi_dict(dict_, func, *args,
```

Comment:
On the other hand, I am not sure why this is not a method of the `BundleDict` class, as it seems that its first input is always an attribute of a BundleDict object.

Reply:
There is one case where it's not called on a BundleDict class. Maybe there could be a more coherent way to do this, but basically we pull an entry out of the bundledict and manipulate it when doing recognition. We use this function for that manipulation. For example, for a given bundle like the left arcuate, to do recognition, we load the ROIs, move them to subject space, and now do distance_transform_edt. This is all done to them as a dict outside the bundledict object as that object is used later in the pipeline (we don't want to manipulate it in place). We also don't want to leave the loaded ROIs attached to the object in memory, we load them, use them, then throw them out, and move on to the next bundle. 

Maybe we should add a 'BundleEntry' class, which this function would accept, but for now they are just their own dicts.

Reply:
Oh - I see: it's also adequately documented in the docstring, but I just missed that. 

I think this is fine, but another option would be to have this still be a class instance, and pass a key as input to select the entry to which this is applied.